CONTRIBUTING.md

```diff
@@ -0,0 +1,338 @@
+## Documentation
+
+A minimal description of the project should be contained in the README.md, then most documentation should generally be contained in a `docs` folder, using [Sphinx](http://www.sphinx-doc.org) (directly or *via* jupyter book) as the documentation generator.
+
+Markdown style should generally follow that rules outlined [markdownlint](https://github.com/markdownlint/markdownlint) and rST should similarly follow [rstcheck](https://github.com/myint/rstcheck), either of which may be added to the pre-commit configuration.
+
+In addition, when writing documentation authors should adhere to the following guidelines:
+
+1. Write **one sentence per line** and otherwise **no manual line wrapping** to make easy to create and review  diffs. All standard editors allow for dynamic line wrapping, and the line length is irrelevant for the rendered documentation in, e.g., HTML or PDF format.
+2. **File and directory names should be alphanumeric** and all lower-case with underscores as word-separators. Example: `entry_points.md`
+3. **Headers must be set in sentence-case**. Example: "Entry points"
+4. Separate paragraphs by one empty line, but not more.
+5. Use the `-` symbol for itemized lists.
+6. Correctly prefix code fences/[code-block directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block) to indicate their usage, e.g.
+
+- Bash shell scripts should use `bash`
+
+  ````md
+  ```bash
+  echo "hi"
+  ```
+  ````
+
+- Bash shell **sessions** (i.e. interactive) should use `console`
+
+  ````md
+  ```console
+  $ echo "hi"
+  ```
+  ````
+
+  Use ``#`` instead of ``$`` to indicate a root prompt.
+
+- Python scripts should use `python`
+  
+  ````md
+  ```python
+  print("hi")
+  ```
+  ````
+
+- Python sessions (e.g. *via* `ipython`) should use `ipython`
+  
+  ````md
+  ```ipython
+  In  [1]: print("hi")
+  Out [1]: "hi"
+  ```
+  ````
+
+(dev/branches)=
+
+## Git Branches
+
+Repositories should use the `master` branch as their primary branch.
+This branch should be "protected" on GitHub and require PR reviews and status checks before merging (see [GitHub branch configuration](https://docs.github.com/en/github/administering-a-repository/configuring-protected-branches)).
+
+Additions to the `master` branch should follow these simple concepts:
+
+- Use feature branches for all new features and bug fixes.
+- Merge feature branches into the master branch using pull requests.
+- Keep a high quality, up-to-date master branch.
+
+It is also advised to name branches by the convention:
+
+- fix/\<issue number\>/\<description\>, e.g. `fix/123/func-error`
+- feature/\<description\>, e.g. `feature/new-options`
+
+(dev/pr_open)=
+
+## Opening a Pull Request
+
+Pull requests should be submitted to `master` early and often!
+
+To improve understanding of pull requests "at a glance", the use of several standardized title prefixes is encouraged:
+
+- **[BRK]** for changes which break existing builds or tests
```

Comment:
don't we want these to have emojis? or do we expect the devs to add them later?

Reply:
Yeh I didn't want to make it too much effort to create a PR for contributors, its something that the devs can do later, since the title will often be changed anyway for the actual commit message.